lib/synapse_pay_rest/http_client.rb

```diff
@@ -194,6 +159,69 @@ def delete(path)
 
     private
 
+    def request_params(method, url:, payload: nil, idempotency_key: nil)
+      request_headers = headers
+      if idempotency_key
+        request_headers.merge!({'X-SP-IDEMPOTENCY-KEY' => idempotency_key})
+      end
+
+      params = {
+        method: method,
+        url: url,
+        headers: request_headers
+      }
+
+      params[:payload] = payload.to_json if payload.present?
+      params
+    end
+
+    # Sends a requests
+    #
+    # @param method [Symbol]
+    # @param url [String]
+    # @param payload [Hash]
+    # @param idempotency_key [String]
+    #
+    # @return [String] Stringified version of the JSON response
+    def perform_request(method:, url:, payload: nil, idempotency_key: nil)
+      RestClient::Request.execute(
+        **request_params(method, url: url, idempotency_key: idempotency_key)
+      )
+    end
+
+    # Sends a request through the VGS proxy
+    #
+    # @param method [Symbol]
+    # @param url [String]
+    # @param payload [Hash]
+    # @param idempotency_key [String]
+    #
+    # @return [String] Stringified version of the JSON response
+    def perform_proxied_request(method:, url:, payload: nil, idempotency_key: nil)
+      request_headers = headers
+
+      if idempotency_key
+        request_headers.merge!({'X-SP-IDEMPOTENCY-KEY' => idempotency_key})
+      end
+
+      options = {
+        proxy: vgs_proxy_config[:host],
+        proxyuserpwd: "#{vgs_proxy_config[:username]}:#{vgs_proxy_config[:password]}",
+        ssl_verifyhost: 5,
```

Comment:
where does this config come from? i'm guessing https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYHOST.html but there is no mention of `5` as a possible value there

Reply:
ups, that was a typo, thanks
I've set the right value `2` (which Typhoeus also documents in [ref](https://github.com/typhoeus/typhoeus/blob/d12276f2c7b228fdf090010b70b9b28463ab72af/README.md#ssl)).